### DISCUSSION TITLE
Rerendering

### DISCUSSION THREAD

**Discussion by rpsutton** [vydimitrov/react-countdown-circle-timer#111] [Q&A]
In some testing, the remainingTime value in the timer's demo code appears to be updating almost 60 times per second? Does this sound correct? How can I make the timer as performant as possible?

**Comment by vydimitrov** [comment]
Hey, I am not sure what is your use case but to me it sounds like premature optimization. In case your `child` function/component does not render 100 other components or does  not do some heavy computation, you should not worry about performance. React does a good job on checking what changed and updates the DOM only when there is a change (like in this case once per second). 

But to answer your question, the component will call the render method approximately every 16ms (on every key frame). If you want to optimize that then you will need to look at the core of the timer, which is a library called `use-elapsed-time`. You will need to change the logic there to save the current elapsed time in `ref` maybe and set it to state once per second.
